File: Fn4IntegradorNFeCommerceWin/Form1.vb
Imports Fn4IntegradorNFeCommerceCtrl

Public Class Form1
    Dim x As New processar_txt

    Public Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click
        x.run()
    End Sub

    Private Sub Button2_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button2.Click
        x.pause()
    End Sub
End Class

File: FN4CancelamentoWin/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using FN4CancelamentoCtl;

namespace FN4CancelamentoWin
{
    public partial class Form1 : Form
    {
        private readonly CancelamentoMonitor _mon = new CancelamentoMonitor();
        
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            try
            {
                _mon.Run();
            }
            catch (Exception exception)
            {
                MessageBox.Show(exception.Message);
            }
        }

        private void button2_Click(object sender, EventArgs e)
        {
            try
            {
                _mon.Pause();
            }
            catch (Exception exception)
            {
                MessageBox.Show(exception.Message);
                throw;
            }
        }
    }
}
